File: D2/D2.php
<?php
// Your code here!
class Submarine{

    public $x = 0;
    public $y = 0;

    function follow_directions($file){
        $lines = file($file);
        foreach ($lines as $line){
           $parts = explode(' ', trim($line);
           print_r($parts)
        }
    }

    function move_it($direction, $n){
        switch($direction){
            case 'forward':
                $this->forward($n);
                break;
            case 'up':
                $this->up($n);
                break;
            case 'down':
                $this->down($n);
                break;
        }
    }

    function forward($n){
        $this->x = $this->x + $n;
    }

    function up($n){
        $this->x = $this->y - $n;
    }

    function down($n){
        $this->x = $this->x + $n;
    }

}


$sub = new Submarine();
$sub->follow_directions('/Users/whaley/Dev/')

File: D2/D2.py
class Submarine:

    x: int = 0
    y: int = 0
    aim: int = 0

    def __init__(self, file):
        [self.move(command) for command in file.readlines()]
        print(self.x * self.y)

    def move(self, command):

        distance: int = int(command.split(' ')[1])

        if 'up' in command:
            self.aim = self.aim - distance

        if 'down' in command:
            self.aim = self.aim + distance

        if 'forward' in command:
            self.x = self.x + distance
            self.y = self.y + (distance * self.aim)


if __name__ == '__main__':
    with open('/Users/whaley/Dev/AOC21/D2/d2_input.txt') as f:
        sub = Submarine(f)
